common/short-hex-str/src/lib.rs

```diff
@@ -98,6 +99,26 @@ fn hex_encode(src: &[u8], dst: &mut [u8]) {
     }
 }
 
+pub trait AsShortHexStr {
+    fn short_str(&self) -> ShortHexStr;
+}
+
+impl AsShortHexStr for [u8; 16] {
+    fn short_str(&self) -> ShortHexStr {
+        const_assert!(16 >= ShortHexStr::SOURCE_LENGTH);
```

Comment:
Could we enforce this at compile time so we can avoid code duplicates by invoking this when this rule is required?

Reply:
Maybe i'm misunderstanding your question but the macro `const_assert` is evaluated at compile time and this check wont be invoked at runtime (although there is another check run in the function below)

Reply:
Oh I see, we have the other check below that actually will never happen.